legal-api/src/legal_api/resources/v2/business/filing_comments.py

```diff
@@ -136,17 +137,22 @@ def _basic_checks(identifier, filing_id, client_request) -> Tuple[dict, int]:
 
     if client_request.method == "GET" and identifier.startswith("T"):
         filing_model = Filing.get_temp_reg_filing(identifier)
-        legal_entity = LegalEntity.find_by_internal_id(filing_model.legal_entity_id)
+        business = None
+        if filing_model.legal_entity_id:
+            business = LegalEntity.find_by_internal_id(filing_model.legal_entity_id)
+        elif filing_model.alternate_name_id:
+            business = AlternateName.find_by_id(filing_model.alternate_name_id)
     else:
-        legal_entity = LegalEntity.find_by_identifier(identifier)
+        business = business_service.fetch_business(identifier)
 
     filing = Filing.find_by_id(filing_id)
 
-    if not legal_entity:
+    if not business:
         return ({"message": f"{identifier} not found"}, HTTPStatus.NOT_FOUND)
 
     # check that filing belongs to this business
-    if not filing or filing.legal_entity_id != legal_entity.id:
+    filing_business_id = filing.legal_entity_id if business.is_legal_entity else filing.alternate_name_id
```

Comment:
you can just use `filing_business_id = filing.legal_entity_id or filing.alternate_name_id`

Reply:
Updated